### PR TITLE
Move DocumentOrShadowRoot.getAnimations into mixin structure

### PR DESCRIPTION
**Pull request by Elchi3**
Spec: https://drafts.csswg.org/web-animations-1/#extensions-to-the-documentorshadowroot-interface-mixin

This PR moves the `api.Document.getAnimations` data to the `api/_mixins/DocumentOrShadowRoot__Document.json` file and adds `api.ShadowRoot.getAnimations` to `api/_mixins/DocumentOrShadowRoot__ShadowRoot.json` 

Discovered while working on re-arranging the content in https://github.com/mdn/content/pull/3029

### CODE REVIEW

File: api/_mixins/DocumentOrShadowRoot__ShadowRoot.json [hunk 1]
```diff
@@ -325,6 +325,288 @@
...
+                    "type": "preference",
+                    "name": "#enable-experimental-web-platform-features"
+                  }
+                ]
+              },
+              {
+                "version_added": "67",
+                "version_removed": "83",
+                "partial_implementation": true,
+                "notes": "Implements an older version of the specification and not supported on <code>ShadowRoot</code>, see <a href='https://crbug.com/828585'>Chromium bug 828585</a>.",
```
Comment (foolip) [inline api/_mixins/DocumentOrShadowRoot__ShadowRoot.json:349]: Any mention of `ShadowRoot` in the notes here will need attention. This one suggests the entry should simply not be here.

File: api/_mixins/DocumentOrShadowRoot__Document.json [hunk 2]
```diff
@@ -396,6 +396,288 @@
...
+                    "type": "preference",
+                    "name": "#enable-experimental-web-platform-features"
+                  }
+                ]
+              },
+              {
+                "version_added": "67",
+                "version_removed": "83",
+                "partial_implementation": true,
+                "notes": "Implements an older version of the specification and not supported on <code>ShadowRoot</code>, see <a href='https://crbug.com/828585'>Chromium bug 828585</a>.",
```
Comment (foolip) [inline api/_mixins/DocumentOrShadowRoot__Document.json:420]: The `ShadowRoot` parts of the notes can be simplified away.

If there was a way to just delete all of this flag data I would be very happy to see that too, really, but only the oldest entry is older than 2 years :/

**Review by foolip — COMMENTED** [review]
I've only skimmed. I'll try comparing this against collector results too at a later time, not today.

**Comment by foolip** [comment]
I found nothing wrong with this data based on mdn-bcd-collector, so I think it's just the notes that need fixing. <sub><sub><sub>... or deleting?</sub></sub></sub>

**Comment by Elchi3** [comment]
Thanks for the quick review! I've made changes to the notes and removed a few support ranges.

**Review by foolip — APPROVED** [review]
I've only skimmed. I'll try comparing this against collector results too at a later time, not today.

**Pull request merged**